### ISSUE TITLE
Tensor observables for fermions

### ISSUE THREAD

**Issue by fretchen** [synqs/pennylane-ls#31]
Running 

```
@qml.qnode(FermionDevice)
def simple_hopping(theta = 0):
    '''
    The circuit that simulates the experiments.
    
    theta ... angle of the hopping
    '''
    # load atoms
    FermionOps.Load(wires=0)
    FermionOps.Load(wires=1)
    
    # let them hop
    FermionOps.Hop(theta, wires=[0,1,2,3])

    # measure the occupation on the right side
    obs = FermionOps.ParticleNumber(0) @ FermionOps.ParticleNumber(1)
    return qml.expval(obs)
```

Leads to the following exception

`Tensor observables not supported on device synqs.fs`

**Comment by TheNewYorkerFlute** [comment]
Not solved yet, but look at the test_fermion_backend.iynb there is an example how to address only certain wires, which works.

![image](https://user-images.githubusercontent.com/39969008/130423569-a7e9dfb9-87e5-42f5-8371-96233ab5966f.png)

**Comment by fretchen** [comment]
Somehow it still always sends back the observation of all wires not only the wires we selected. And I am not sure how to obtain the expval there. But no errors for the moment.

**Comment by TheNewYorkerFlute** [comment]
Yes, the particle number operator observable sums over all wires which are selected. 

![image](https://user-images.githubusercontent.com/39969008/130778470-6359579c-550f-41e3-a97b-a1551790ff5a.png)

This is wrongly implemented and has to be fixed.

**Comment by TheNewYorkerFlute** [comment]
Okay, one can now select the observables

![image](https://user-images.githubusercontent.com/39969008/130793136-e2ad48a0-b9ae-4eab-b010-19a5cbc399bd.png)

**Comment by fretchen** [comment]
I think that this looks good. I implemented this into the notebook in b90d4b5.
